### PR TITLE
Add more invoice history actions

### PR DESCRIPTION
**Pull request by apoorv1316**
Notion
https://www.notion.so/saeloun/More-details-needs-to-be-displayed-in-the-view-history-panel-d005aac54b3040c697c092a224a1d3fb?d=1cfd0ea5f022496c9a3118c62ced62e7

What
- Show the email addresses of users for all actions
- Capture manual payment event
- Capture stripe payment event
- Add email addresses of all recipients on send_invoice action

Loom
https://www.loom.com/share/87b573334bde43b2986094469f631cac

### CODE REVIEW

File: app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts [hunk 1]
```diff
@@ -99,19 +101,26 @@ export const createData = (rawData, company) => {
...
       case "send_invoice": {
-        const message = `${data.userName} sent invoice`;
+        let message = `${userDisplayName}<${userEmail}> sent invoice`;
         const time = calculateTime(
           data.createdAt,
           company.date_format,
           company.timezone
         );
+
+        if (Array.isArray(data.emails) && data.emails.length > 0) {
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts:111]: Move this logic to a function and call it here.

File: app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts [hunk 2]
```diff
@@ -99,19 +101,26 @@ export const createData = (rawData, company) => {
         break;
       }
       case "send_invoice": {
-        const message = `${data.userName} sent invoice`;
+        let message = `${userDisplayName}<${userEmail}> sent invoice`;
         const time = calculateTime(
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts:105]: each case has the same return logic. Move the common logic to function and call it inside the each case

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by apoorv1316** [comment]
@prasanthchaduvula Please review

File: app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts [hunk 3]
```diff
@@ -46,112 +46,122 @@ const calculateTime = (date, dateFormat, timezone) => {
...
-        const message = "Partial payment successful";
-        const time = calculateTime(
+        const partialPaymentMessage = "Partial payment successful";
+        const partialPaymentTime = calculateTime(
           data.createdAt,
           company.date_format,
           company.timezone
         );
-        Data.push({ message, time });
+        Data.push({ message: partialPaymentMessage, time: partialPaymentTime });
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts:141]: Can't we use `processInvoiceEvent` here

File: app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts [hunk 4]
```diff
@@ -46,112 +46,122 @@ const calculateTime = (date, dateFormat, timezone) => {
...
         }`;
 
-        const time = calculateTime(
+        const cancellationTime = calculateTime(
           data.createdAt,
           company.date_format,
           company.timezone
         );
-        Data.push({ message, time });
+        Data.push({ message: cancellationMessage, time: cancellationTime });
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts:154]: ditto

File: app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts [hunk 5]
```diff
@@ -46,112 +46,122 @@ const calculateTime = (date, dateFormat, timezone) => {
...
-        const message = "Payment Failed";
-        const time = calculateTime(
+        const failedPaymentMessage = "Payment Failed";
+        const failedPaymentTime = calculateTime(
           data.createdAt,
           company.date_format,
           company.timezone
         );
-        Data.push({ message, time });
+        Data.push({ message: failedPaymentMessage, time: failedPaymentTime });
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/ViewHistory/utils.ts:164]: ditto

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Review by supriya3105 — APPROVED** [review]
The requested functionality is implemented. However, will require testing to confirm the behaviour .

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**